jbmc/src/java_bytecode/select_pointer_type.h

```diff
@@ -19,6 +19,7 @@
 
 typedef std::unordered_map<irep_idt, std::vector<reference_typet>>
   generic_parameter_specialization_mapt;
+typedef std::set<irep_idt> generic_parameter_recursion_trackingt;
```

Comment:
I'm not sure this typedef is helpful `std::set<irep_idt>` is shorter and I find it more informative

Reply:
The idea is to make is to make it easy to change, if desired.